### PR TITLE
Working on app functionality

### PR DESCRIPTION
**Pull request by Dominic-dul**
-When player press on 'Occupy' in PinInfoPage, the points worth of location are added to his personal points and it displays in the ProfilePage
-When player presses 'More Info' on location alert, when pin is clicked, the locationsVisited increases by 1 and it shows in the ProfilePage
-When player presses 'Occupy' in PinInfoPage, the owner of the current location changes and that location is assigned to the player, who occupied it
-The updated location database then displays this Occupied location as his own, when 'Show My Locations' button is clicked
-Added 'Points' and 'Player ID' labels in Leaderboard for clarity
-PinInfoPage now gets the whole location as a constructor parameter because it's more convenient

### CODE REVIEW

File: EncounterMeApp/EncounterMeApp/Views/MapPage.xaml.cs [hunk 1]
```diff
@@ -63,11 +65,13 @@ public async void DisplayExistingPins(Lazy<Task<List<MyLocation>>> list)
                 {
                     args.HideInfoWindow = true;
                     string pinName = ((Pin)s).Label;
-                    string action = await DisplayActionSheet(pinName, "Cancel", "Occupy", $"COORDS: {((Pin)s).Position.Latitude},{((Pin)s).Position.Longitude}",
+                    string action = await DisplayActionSheet(pinName, "OK", "Cancel", $"COORDS: {((Pin)s).Position.Latitude},{((Pin)s).Position.Longitude}", //Need to remove the 'OK' button
                         $"Points: {location.points}", $"Owner: {location.owner}", "More info");
                     if (action == "More info")
                     {
-                        _ = Navigation.PushAsync(new PinInfoPage(pinName, location.owner, location.points));
+                        App.player.LocationsVisited += 1;
```
Comment (PijusPale) [inline EncounterMeApp/EncounterMeApp/Views/MapPage.xaml.cs:72]: For LocationsVisited we should check if the person is near the actual location, not just add +1 when person check for information

**Review by PijusPale — APPROVED** [review]
Everything else seems good, no major changes but they definitely help us with the whole app flow. Great job!

**Review by MartynasAnn — APPROVED** [review]
Everything seems good, repeating what Pijus has said, we need actual checks for GPS data of the user to make sure they're at the right location to perform actions.

File: EncounterMeApp/EncounterMeApp/Views/MapPage.xaml.cs (hunk 1 above)
Comment (Dominic-dul) [reply EncounterMeApp/EncounterMeApp/Views/MapPage.xaml.cs:72]: In the future we will add a check if player is near the place and then enable the possibility to occupy that place of his choice

**Pull request merged**